### PR TITLE
[Distributed] Accept model names

### PR DESCRIPTION
**Pull request by kwen2501**
### Change 1
Add argparser to accept model name and other arguments (e.g. PP degree).

```
$ torchrun dist_run.py -h
usage: dist_run.py [-h] [--pp PP] {llama2-7b-chat,llama3}

positional arguments:
  {llama2-7b-chat,llama3}
                        Name of the model to load

options:
  -h, --help            show this help message and exit
  --pp PP               Pipeline parallel degree
```

Supported model names:
```
llama2-7b-chat
llama3
```
(Following model aliases in [models.json](https://github.com/pytorch/torchchat/blob/main/torchchat/model_config/models.json))

### Change 2
Renamed variable `hf_model_name` to `distribution`.

A distribution looks like this:
`meta-llama/Llama-2-7b-chat-hf`
`meta-llama/Meta-Llama-3-8B-Instruct`

### Change 3
Fix `output` undefined error when PP = 1.

### CODE REVIEW

File: dist_run.py [hunk 1]
```diff
@@ -182,7 +195,8 @@ def main():
 
     # Distribute model on TP mesh
     model.distribute(tp_mesh)
-    logger.info(f"Model: {model}")
+    if rank == 0:
```
Comment (lessw2020) [inline dist_run.py:198]: nit: already filtering based on ranks in the dist_run.sh launcher rather than having express rank based logging in the code.

**Review by lessw2020 — APPROVED** [review]
looks good! 
I also have toml config supporting code in distributed/config_manager.py that I had integrated earlier, so we should add in both cmd line and toml support ultimately (similar to titan where you can use either).

**Comment by Jack-Khuu** [comment]
Rebasing on main to pick up CI fix

File: dist_run.py (hunk 1 above)
Comment (kwen2501) [reply dist_run.py:198]: Yeah, it is a hard choice.. I don't know if regular users use filter (I never remember that option's name) and want to have a minimal command line work.

**Comment by kwen2501** [comment]
> Rebasing on main to pick up CI fix

Thanks @Jack-Khuu !

**Pull request merged**